src/blob_storage.cc

```diff
@@ -8,22 +8,13 @@ namespace titandb {
 
 Status BlobStorage::Get(const ReadOptions& options, const BlobIndex& index,
                         BlobRecord* record, PinnableSlice* value) {
-  auto sfile = FindFile(index.file_number).lock();
-  if (!sfile)
-    return Status::Corruption("Missing blob file: " +
-                              std::to_string(index.file_number));
-  return file_cache_->Get(options, sfile->file_number(), sfile->file_size(),
-                          index.blob_handle, record, value);
+  return file_cache_->Get(options, index.file_number, index.blob_handle, record,
```

Comment:
It will reduce the cost of a lock, but will add a cost to get file size from handle, is it cheaper?

Reply:
Agree, can you give some numbers?

Reply:
Hard to tell, because it depends on how long the mutex is held by others. As the mutex is shared by multiple fields, it's better to use an operation not affected by the mutex.

At least, rocksdb does this way https://github.com/tikv/rocksdb/blob/6.29.tikv/db/blob/blob_file_reader.cc#L97

Reply:
OK. After reading the code again, I think the difference is that before this PR, the lock is accquired for every call. 
While for this PR, the getting file size only happens when cache miss. So overall it should be better.